src/main/java/org/seqdoop/hadoop_bam/FastaInputFormat.java

```diff
@@ -54,98 +54,119 @@
-	    FSDataInputStream fis = fs.open(path);
-	    byte[] buffer = new byte[1024];
-
-	    long byte_counter = 0;
-	    long prev_chromosome_byte_offset = 0;
-	    boolean first_chromosome = true;
-
-	    for(int j = 0; j < splits.size(); j++) {
-		FileSplit origsplit = (FileSplit)splits.get(j);
-
-		while(byte_counter < origsplit.getStart()+origsplit.getLength()) {
-		    long bytes_read = fis.read(byte_counter, buffer, 0, (int)Math.min(buffer.length,
-										      origsplit.getStart()+origsplit.getLength()- byte_counter));
-		    //System.err.println("bytes_read: "+Integer.toString((int)bytes_read)+" of "+Integer.toString(splits.size())+" splits");
-		    if(bytes_read > 0) {
-			for(int i=0;i<bytes_read;i++) {
-			    if(buffer[i] == (byte)'>') {
-				//System.err.println("found chromosome at position "+Integer.toString((int)byte_counter+i));
-				
-				if(!first_chromosome) {
-				    FileSplit fsplit = new FileSplit(path, prev_chromosome_byte_offset, byte_counter + i-1 - prev_chromosome_byte_offset, origsplit.getLocations());
-				    //System.err.println("adding split: start: "+Integer.toString((int)fsplit.getStart())+" length: "+Integer.toString((int)fsplit.getLength()));
-				    
-				    newSplits.add(fsplit);
+		// Note: We generate splits that correspond to different sections in the FASTA
+		// input (which here are called "chromosomes", delimited by '>' and
+		// followed by a single line description.
+		// Some locality is preserved since the locations are formed from the input
+		// splits, although no special attention is given to this issues (FASTA files
+		// are assumed to be smallish).
+		// The splits are generated on the client. In the future the split generation
+		// should be only performed once and an index file stored inside HDFS for
+		// peformance reasons. Currently this is not attempted (again: FASTA files
+		// aren't all that big).
+
+		// we first make sure we are given only a single file
+
+		List<InputSplit> splits = super.getSplits(job);
+
+		// first sort by input path
+		splits.sort((a, b) -> {
+			FileSplit fa = (FileSplit) a, fb = (FileSplit) b;
+			return fa.getPath().compareTo(fb.getPath());
+		});
+
+		for (int i = 0; i < splits.size()-1; i++) {
+			FileSplit fa = (FileSplit)splits.get(i);
+			FileSplit fb = (FileSplit)splits.get(i+1);
+
+			if(fa.getPath().compareTo(fb.getPath()) != 0)
+				throw new IOException("FastaInputFormat assumes single FASTA input file!");
+		}
+
+		// now we are sure we only have one FASTA input file
+
+		final List<InputSplit> newSplits = new ArrayList<>(splits.size());
+		FileSplit fileSplit = (FileSplit)splits.get(0);
+		Path path = fileSplit.getPath();
+
+		FileSystem fs = path.getFileSystem(job.getConfiguration());
+		FSDataInputStream fis = fs.open(path);
+		byte[] buffer = new byte[1024];
+
+		long byte_counter = 0;
+		long prev_chromosome_byte_offset = 0;
+		boolean first_chromosome = true;
+
+		for(int j = 0; j < splits.size(); j++) {
```

Comment:
... rather than add the same comment everywhere, can whitespace after keyword before paren be made consistent?